app/src/main/java/com/github/sdp_begreen/begreen/activities/SignInActivity.kt

```diff
@@ -60,8 +66,8 @@ class SignInActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_sign_in)
 
+        // Find the sign-in button and initialize firebaseAuth
         signInGoogleLayout =findViewById(R.id.signInGoogleLayout)
-
         firebaseAuth = FirebaseAuth.getInstance()
 
         // When the google sign in image is clicked, the google sign in page appears
```

Comment:
Here also you should create a separate private helper function that handles the behavior of the UI component